### PR TITLE
week-9-c

### PR DESCRIPTION
**Pull request by SMVIAL**
Exploring the fragility of LQG controls schemes and how that provides a natural path to Robust control theory.

### CODE REVIEW

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb [hunk 1]
```diff
@@ -0,0 +1,363 @@
+{
```
Comment (StewartLamon) [inline bite-size-notes/week-8-c-LQG-stability-smvial.ipynb:1]: I think this is a great review and explanation on the ideas presented in John Doyles paper on LQG stability margins and I like the use of code as part of the explanation. 

I wonder if it wouldn't be too much trouble to add a nyquist or bode plot to visually show the stability margins.

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+    "Boyle somehow derived an analytic solution for the control gain (presumably using the time he saved for writing an abstract).  He came up with:\n",
+    "$$\n",
+    "K = (2 + \\sqrt{4+q})\\begin{bmatrix}\n",
+    "1 & 1 \n",
+    "\\end{bmatrix}\n",
+    "$$\n",
+    "\n",
+    "The above code computes and displays this analytic solution for comparison to our numerical soluton (given same q).  We can confirm that we are on the right track. \n",
+    "\n",
+    "Next, we will do basically the same thing, but on the estimator side of our LQR.  We will solve an ARE for the steady state $\\Sigma$, and then solve for our Kallman filter gain, $L = \\Sigma C^T S_v^{-1}$. The ARE in this case is\n",
```
Comment (StewartLamon) [inline bite-size-notes/week-8-c-LQG-stability-smvial.ipynb:171]: I don't think sigma or S were defined though they presumably are the same values of Q and R.

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+    "d & 0 & -d-f & 1-f\\\\\n",
+    "\\end{bmatrix} \n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Exposing Instability\n",
```
Comment (StewartLamon) [inline bite-size-notes/week-8-c-LQG-stability-smvial.ipynb:289]: There are a few sections where the equation gets cut off of the page. Such as the characteristic polynomial in the exposing instability section and the equations after. I suggest trying display the equation on multiple lines. The formatting on the two coefficient terms is a little confusing at first glance as well. It looked like a mistake until I realized that the inequalities were aligned. There are extra $$ characters.

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+    "$$\n",
+    "\n",
+    "\\begin{align*}\n",
+    "1 + (1-m)df &> 0 & \\qquad d+f - 4 + 2df(m-1) &> 0 \\\\\n",
+    "m &< 1 + \\frac{1}{df}  & \\qquad m &> \\frac{4 - d - f}{2df} + 1  \\\\\n",
+    "\\end{align*}\n",
+    "\n",
+    "$$\n",
+    "\n",
+    "When d and f (or equivalently, q and $\\sigma_w$) become sufficiently large, $ m \\gtrsim 1$ AND $m \\lesssim 1$.  Since, $g$ and $f$ cannot actually go to infinity, the system will technically be stable as long as it exhibits a gain of exactly one.  However, if it deviates slightly in either direction, it will become unstable! There is the rub: **An LQG system has no guarunteed stability margin.  It may operate arbitrarily close to a pole.** \n",
```
Comment (StewartLamon) [inline bite-size-notes/week-8-c-LQG-stability-smvial.ipynb]: I don't know if i understand what is meant by operating arbitrarily close to a pole. Perhaps you meant close to an unstable pole?

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb (hunk 2 above)
Comment (SMVIAL) [reply bite-size-notes/week-8-c-LQG-stability-smvial.ipynb:171]: These are defined in the "defining the system" section.  However, you are correct, $S_w$ and $S_v$ correspond to what we have been calling Q and R in class.  I just disliked the overloading of Q and R as both cost function weights and random variable covariances. $\sigma_w$ and $\sigma_v$ are the standard deviation of individual elements of the W and V random variable vectors.

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb (hunk 1 above)
Comment (SMVIAL) [reply bite-size-notes/week-8-c-LQG-stability-smvial.ipynb:1]: I think adding a bode plot would be a fabulous idea.  Unfortunately, I am unable to.  This is my first controls theory class, and they were not covered.  I have received an introduction to them while researching this topic, and have found a textbook co-authored by Steve Brunton (UW ME department) and Nathan Kutz (UW AMATH department) that transitions into robust control theory and teaches bode plots.  It is source 2 of these notes.

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb (hunk 3 above)
Comment (SMVIAL) [reply bite-size-notes/week-8-c-LQG-stability-smvial.ipynb:289]: I have cleaned up the long polynomial by splitting it into multiple lines.  And I removed the extra dollar signs (just a formatting issue) from the inequalities.  Thanks.

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb (hunk 4 above)
Comment (SMVIAL) [reply bite-size-notes/week-8-c-LQG-stability-smvial.ipynb]: This is my fault.  I should have been more careful using terminology from robust control theory when I have not yet learned it.  I understand that in this field, we refer to unstable fixed points as poles.  That was not what I should have said.  What I should have said is that the system dynamics can operate arbitrarily close to having positive eigenvalues - and thus becoming unstable. 

I have changed the language accordingly.  Good editorial input.

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb (hunk 1 above)
Comment (SMVIAL) [reply bite-size-notes/week-8-c-LQG-stability-smvial.ipynb:1]: I noticed that about halfway through the notes I accidentally started referring to Doyle as Boyle... as in Jake Peralta's best friend/partner.  You were just going to let me go with that?! Like letting me go give a speech with something in my teeth.  That's cold.

File: bite-size-notes/week-8-c-LQG-stability-smvial.ipynb (hunk 2 above)
Comment (SMVIAL) [reply bite-size-notes/week-8-c-LQG-stability-smvial.ipynb:171]: Aaah.  I see what you may have meant.  I introduce these variables, but I do not explicitly define them (except in code). Noted.

**Pull request merged**